From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Mon, 5 Apr 2010 11:43:18 +0000 (+0400)
Subject: ve: fix ve task state percpu counters
X-Git-Url: http://git.openvz.org/?p=linux-2.6.27-openvz;a=commitdiff_plain;h=762c78b5d7f1c293133651ac2096179f161ef0c0;hp=f83e9da10c15dcb6532a1d1bf3e46a4758c7ad97

ve: fix ve task state percpu counters

Counters overlap detection for ve tasks in running/uninterraprible/iowait state
was broken due to type mismatch:
nr_{running/unin..e/iowait}_ve() uses _long_ for summing _int_ percpu counters.

As result, it broke ve loadavg calculation after first int overlap.

This patch expand all this percpu counters to unsigned long.

http://bugzilla.openvz.org/show_bug.cgi?id=1396

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/include/linux/ve.h b/include/linux/ve.h
index 7d12746..821a2be 100644
--- a/include/linux/ve.h
+++ b/include/linux/ve.h
@@ -129,9 +129,9 @@ struct ve_cpu_stats {
 	cycles_t	strt_idle_time;
 	cycles_t	used_time;
 	seqcount_t	stat_lock;
-	int		nr_running;
-	int		nr_unint;
-	int		nr_iowait;
+	unsigned long	nr_running;
+	unsigned long	nr_unint;
+	unsigned long	nr_iowait;
 	cputime64_t	user;
 	cputime64_t	nice;
 	cputime64_t	system;
